code/datums/soundOutput.dm

```diff
@@ -167,18 +232,9 @@
 
 /datum/soundOutput/proc/on_mob_moved(datum/source, atom/oldloc, direction, Forced)
 	SIGNAL_HANDLER //COMSIG_MOVABLE_MOVED
+	update_sounds()
 	update_area_environment()
 
-/datum/soundOutput/proc/on_mob_logout(datum/source)
-	SIGNAL_HANDLER //COMSIG_MOB_LOGOUT
-	UnregisterSignal(owner.mob, list(COMSIG_MOVABLE_MOVED, COMSIG_MOB_LOGOUT))
-
-/datum/soundOutput/proc/on_client_mob_logged_in(datum/source, mob/new_mob)
-	SIGNAL_HANDLER //COMSIG_CLIENT_MOB_LOGGED_IN
-	RegisterSignal(owner.mob, COMSIG_MOVABLE_MOVED, PROC_REF(on_mob_moved))
-	RegisterSignal(owner.mob, COMSIG_MOB_LOGOUT, PROC_REF(on_mob_logout))
-	update_mob_environment_override()
```

Comment:
Still need the `update_mob_environment_override()` part of this. When a mob's environment override changes (currently just ear-ringing when in crit) it informs the soundOutput of the change. When the client's *mob* changes, however, soundOutput doesn't know unless we tell it. So for instance someone in crit that gets ghosted will have ear-ringing unless it knows to update, such as from `COMSIG_CLIENT_MOB_LOGGED_IN`.